modules/internal/ChapelArray.chpl

```diff
@@ -2346,6 +2346,15 @@ module ChapelArray {
       return _value.dsiIteratorYieldsLocalElements();
     }
 
+    operator :(val: _domain, type t: string) {
+      use Reflection;
+      if canResolveMethod(val._value, "dsiToString") {
+        return val._value.dsiToString();
```

Comment:
Should `dsiToString` be added to the dsi tech note?

Reply:
Hmm, your question is making me realize that this should really be a `.doi...` routine given that it's optional (and may not make sense for something like a sparse or associative domain where the generated code size of converting to a string is likely to dwarf the general `stringify()` routine.  And then the question still holds, except that it seems we don't document many of the doi routines in that file...?  [mostly, I feel like that file is waiting for an interface to replace it]

Reply:
(I did end up adding it, though from an account where I didn't have DCO signed up, so need to clean up.  While adding more testing, I also found some pre-existing bugs in the range cast code which were easy to fix, but will require a re-test.